File: CloudflareDatabase.py
import sqlite3
from urllib.parse import urlparse

def handle_request(request):
    conn = sqlite3.connect('cloudflare.db')
    cursor = conn.cursor()

    if request.startswith("/setvar"):
        parts = request.split('/')[1:]
        if len(parts) < 2:
            print("Error: Key or Value missing")
            return
        key, value = parts[0], parts[1]
        cursor.execute("INSERT OR REPLACE INTO variables (key, value) VALUES (?, ?)", (key, value))
        conn.commit()
        print(f"Variable '{key}' set to '{value}'.")

    elif request.startswith("/getvar"):
        parts = request.split('/')[1:]
        if len(parts) < 1:
            print("Error: Key missing")
            return
        key = parts[0]
        cursor.execute("SELECT value FROM variables WHERE key = ?", (key,))
        row = cursor.fetchone()
        if row:
            print(row[0])
        else:
            print("Variable not found")
    else:
        print("Hello from Cloudflare D1!")

    conn.close()

# Example usage
handle_request("/setvar/myKey/myValue")
handle_request("/getvar/myKey")

File: CloudflareDatabase.php
<?php
function handleRequest($request) {
    $db = new SQLite3('cloudflare.db');

    if (strpos($request, '/setvar') !== false) {
        // Set variable
        preg_match('/\/setvar\/([^\/]+)\/([^\/]+)/', $request, $matches);
        if (count($matches) < 3) {
            echo "Error: Key or Value missing";
            return;
        }

        $key = $matches[1];
        $value = $matches[2];

        $stmt = $db->prepare('INSERT OR REPLACE INTO variables (key, value) VALUES (:key, :value)');
        $stmt->bindValue(':key', $key, SQLITE3_TEXT);
        $stmt->bindValue(':value', $value, SQLITE3_TEXT);
        $stmt->execute();

        echo "Variable '$key' set to '$value'.";
    }
    elseif (strpos($request, '/getvar') !== false) {
        // Get variable
        preg_match('/\/getvar\/([^\/]+)/', $request, $matches);
        if (count($matches) < 2) {
            echo "Error: Key missing";
            return;
        }

        $key = $matches[1];
        $stmt = $db->prepare('SELECT value FROM variables WHERE key = :key');
        $stmt->bindValue(':key', $key, SQLITE3_TEXT);
        $result = $stmt->execute();

        $row = $result->fetchArray(SQLITE3_ASSOC);
        if ($row) {
            echo $row['value'];
        } else {
            echo "Variable not found";
        }
    } else {
        echo "Hello from Cloudflare D1!";
    }
}

// Example Usage
handleRequest("/setvar/myKey/myValue");
handleRequest("/getvar/myKey");
?>
